docs/keto/modeling/create-permission-model.mdx

```diff
@@ -0,0 +1,377 @@
+
+In Ory Permissions, objects and subjects you identified are [namespaces](../concepts/namespaces.mdx). They are declared as classes
+in the Ory Permission Language. The convention is to name these classes like TypeScript classes, starting with a capital letter
+and and in the singular form, for example `Document` instead of `documents`.
+
+:::
+
+### Determine relationships between objects
+
+Read through the initial assumptions point by point to determine the list of relationships:
+
+- The goal is to model `user` access to `documents` such that `users` can be the `owner`, `editor`, or `viewer` of a `document`.
+
+  ```mdx-code-block
+  <Mermaid
+    chart={`
+  graph LR
+      Document -->|owner| User
+      Document -->|editor| User
+      Document -->|viewer| User
+  `}
+  />
+  ```
+
+- Every owner is also an editor of a `document` and every editor is also a viewer of a `document`. Furthermore, `documents` can be
+  put into a hierarchy of `folders`. If a user can view the parent `folder`, they can also view all `folders` and `documents` the
+  parent `folder` contains.
+
+  ```mdx-code-block
+  <Mermaid
+    chart={`
+  graph LR
+      Document -->|parent| Folder
+      Folder -->|parent| Folder
+      Folder -->|owner| User
+      Folder -->|editor| User
+      Folder -->|viewer| User
+  `}
+  />
+  ```
+
+This is the complete matrix of relationships presented on a single diagram:
+
+```mdx-code-block
+<Mermaid
+  chart={`
+graph LR
+    Document -->|owner| User
+    Document -->|editor| User
+    Document -->|viewer| User
+    Document -->|parent| Folder
+    Folder -->|parent| Folder
+    Folder -->|owner| User
+    Folder -->|editor| User
+    Folder -->|viewer| User
+`}
+/>
+```
+
+### Define relation tuples in the OPL
+
+Next, add each relation to the permission config.
+
+:::tip
+
+In Ory Permissions, relationships are declared inside the corresponding class in the Ory Permission Language. Since you store
+arbitrary tuples in Ory Permissions, you always have many-to-many relations between these objects. To reflect this in OPL, the
+convention is to pluralize the relation name, for example, `viewers` instead of `viewer`.
+
+:::
+
+```ts title="permissions-v2.ts"
+import { Namespace, Context } from "@ory/keto-namespace-types"
+
+class User implements Namespace {}
+
+class Document implements Namespace {
+  // highlight-start
+  related: {
+    owners: User[]
+    editors: User[]
+    viewers: User[]
+    parents: Folder[]
+  }
+  // highlight-end
+}
+
+class Folder implements Namespace {
+  // highlight-start
+  related: {
+    owners: User[]
+    editors: User[]
+    viewers: User[]
+    parents: Folder[]
+  }
+  // highlight-end
+}
+```
+
+### List permissions to check for each object
+
+When you perform an action on behalf of a user, you should check for a specific permission, such as `view` or `edit`, instead of a
+relationship, such as `owner`. Therefore, each permission is derived from relationships and other permissions.
```

Comment:
I don't understand where does that `Therefore, each permission is derived from relationships and other permissions.` in the context of this paragraph ;( Is there any way you could rephrase it to make it more obvious?

Reply:
```suggestion
relationship, such as `owner`. The concrete permission is then still checked against the relationships. For example, if `owners` can `view` a file, and the `view` permission is checked, then the `owners` relation is looked up in the relationships database.
```

Reply:
Does that make it clearer?

Reply:
Yes, I think it's better now.